src/plugins/janus_sip.c

```diff
@@ -6625,14 +6652,27 @@ static int janus_sip_allocate_local_ports(janus_sip_session *session, gboolean u
 			session->media.pipefd[1] = -1;
 		}
 	}
+	/* Determine the address family to use */
+	if(ipv6_disabled &&
+		!janus_network_address_is_null(&janus_network_local_media_ip) &&
+		janus_network_local_media_ip.family == AF_INET6) {
+		JANUS_LOG(LOG_ERR, "Error setting address for media sockets, IPv6 is disabled and local media address is IPv6...\n");
```

Comment:
I don't like this action failing at runtime. It looks like a check that can be done when thew plugin is loaded, that is after the `janus_network_local_media_ip` property is filled in and we know if `ipv6_disabled` is `FALSE`. Better to have the plugin not load at all, if we already know it will never allocate sockets for media.

Reply:
Makes sense. Moved the failure to the config loading time.